src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -36,6 +36,116 @@
 */
 public with sharing class BGE_BatchGiftEntry_UTIL {
 
+    /*******************************************************************************************************
+    * @description Returns the relevant Batch fields for soql
+    * @return list of field API names
+    */
+    public static List<String> getBatchFieldNames() {
+        return new List<String>{
+            'Id',
+            'Name',
+            'Active_Fields__c',
+            'Batch_Description__c',
+            'Batch_Process_Size__c',
+            'Contact_Matching_Rule__c',
+            'Donation_Matching_Behavior__c',
+            'Donation_Matching_Implementing_Class__c',
+            'Donation_Matching_Rule__c',
+            'Donation_Date_Range__c',
+            'Expected_Count_of_Gifts__c',
+            'Expected_Total_Batch_Amount__c',
+            'Post_Process_Implementing_Class__c',
+            'Process_Using_Scheduled_Job__c',
+            'RequireTotalMatch__c',
+            'Run_Opportunity_Rollups_while_Processing__c'
+
+        };
+    }
+
+    /*******************************************************************************************************
+    * @description Checks to see whether the running user has permission to edit a given list of fields
```

Comment:
Would we ever want to check read permission independent of edit? For use as a utility method would we want that as an optional argument potentially?